### PR TITLE
Fix building on MSYS2

### PR DESCRIPTION
**Pull request by iahung**
The patch is based on the work of @Biswa96 of the MSYS2 project:

https://github.com/msys2/MINGW-packages/discussions/16526

### CODE REVIEW

**Comment by Biswa96** [comment]
This may break existing build workflow. I have already said this may not be appropriate for upstream. Also the mingw package is not complete--the paths do not follow typical *nix-like FHS.

**Comment by Biswa96** [comment]
This may break existing build workflow. Also the mingw package is not complete--the paths do not follow typical *nix-like FHS.

File: mk/cmake/SuperTux/ProvideSquirrel.cmake [hunk 1]
```diff
@@ -41,7 +41,7 @@ else()
     -DINSTALL_INC_DIR=include
     -DCMAKE_POSITION_INDEPENDENT_CODE=ON)
 
-  if(WIN32)
+  if(MSVC)
```
Comment (Biswa96) [inline mk/cmake/SuperTux/ProvideSquirrel.cmake:44]: This is not required. cmake can find squirrel from the package.

**Comment by iahung** [comment]
I tested both the nightly built using MSVC provided by SuperTux and my own version built using MSYS2. The MSYS2 version has weird bug in game play compared to the official nightly. I have no idea why it's this way.

**Pull request closed without merging**